src/Formula/Parser.elm

```diff
@@ -185,6 +197,34 @@ term =
             )
 
 
+substitution : Parser Term.Substitution
+substitution =
+    Parser.sequence
+        { start = "{"
+        , separator = ","
+        , end = "}"
+        , spaces = spaces
+        , item = substPair
+        , trailing = Forbidden
+        }
+        |> map Dict.fromList
+
+
+makeSubstPair : String -> Term -> ( String, Term )
+makeSubstPair s t =
+    ( s, t )
```

Comment:
This function is built in as [Tuple.pair](https://package.elm-lang.org/packages/elm/core/latest/Tuple#pair). Also, such simple functions are usually written as closures: `\s t -> (s, t)`.